initial commit

AleksandrPok opened the pull request.

AleksandrPok committed:
initial commit

less08 commented on pom.xml:
```diff
@@ -8,5 +8,68 @@
 
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
```
We don't really need this dependency, since we did't write any tests

less08 commented on src/main/java/injectorpractice/model/Car.java:
```diff
@@ -0,0 +1,49 @@
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(long id, String model, Manufacturer manufacturer, List<Driver> drivers) {
```
https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/03_dao_service_checklist.html
last point

less08 commented on src/main/java/injectorpractice/model/Car.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.List;
+
+public class Car {
+    private long id;
```
Use Wrapper

less08 commented on src/main/java/injectorpractice/model/Driver.java:
```diff
@@ -0,0 +1,37 @@
+package injectorpractice.model;
+
+public class Driver {
+    private long id;
```
Use Wrapper

less08 commented on src/main/java/injectorpractice/model/Manufacturer.java:
```diff
@@ -0,0 +1,42 @@
+package injectorpractice.model;
+
+public class Manufacturer {
+    private long id;
```
Wrapper

AleksandrPok committed:
deleted unnecessary dependencies;
removed List<Driver> from Car constructor;
replaced primitives long with wrappers Long

Sofasmile commented on pom.xml:
```diff
@@ -8,5 +8,60 @@
+            </plugins>
+        </pluginManagement>
+    </build>
 </project>
```
let's add empty line at the end of the file

Sofasmile commented on src/main/java/injectorpractice/dao/ManufacturerDaoImpl.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream().filter(m -> m.getId().equals(id)).findFirst();
```
```suggestion
        return Storage.manufacturers.stream()
                .filter(m -> m.getId().equals(id))
                .findFirst();
```

Sofasmile commented on src/main/java/injectorpractice/dao/ManufacturerDaoImpl.java:
```diff
@@ -0,0 +1,35 @@
+    public Manufacturer update(Manufacturer manufacturer) {
+        IntStream.range(0, Storage.manufacturers.size())
+                .filter(i -> (Storage.manufacturers.get(i).getId().equals(manufacturer.getId())))
+                .findFirst().ifPresent(i -> Storage.manufacturers.set(i, manufacturer));
```
```suggestion
                .findFirst()
                .ifPresent(i -> Storage.manufacturers.set(i, manufacturer));
```

Sofasmile commented on src/main/java/injectorpractice/model/Car.java:
```diff
@@ -0,0 +1,50 @@
+        this.id = id;
+        this.model = model;
+        this.manufacturer = manufacturer;
+        this.drivers = new ArrayList<>();
```
```suggestion
        drivers = new ArrayList<>();
```

Sofasmile commented on src/main/java/injectorpractice/model/Car.java:
```diff
@@ -0,0 +1,50 @@
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(Long id, String model, Manufacturer manufacturer) {
```
should pass id via constructor?

Sofasmile commented on src/main/java/injectorpractice/model/Car.java:
```diff
@@ -0,0 +1,50 @@
+        this.drivers = new ArrayList<>();
+    }
+
+    public long getId() {
```
```suggestion
    public Long getId() {
```

Sofasmile commented on src/main/java/injectorpractice/model/Driver.java:
```diff
@@ -0,0 +1,37 @@
+    private String name;
+    private String licenceNumber;
+
+    public Driver(Long id, String name, String licenceNumber) {
```
the same

Sofasmile commented on src/main/java/injectorpractice/service/ManufacturerServiceImpl.java:
```diff
@@ -0,0 +1,40 @@
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+    @Inject
+    ManufacturerDao manufacturerDao;
```
how about access modifier?

Sofasmile commented on src/main/java/injectorpractice/service/ManufacturerServiceImpl.java:
```diff
@@ -0,0 +1,40 @@
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        manufacturerDao.create(manufacturer);
+        return manufacturer;
```
```suggestion
        return manufacturerDao.create(manufacturer);
```

Sofasmile commented on src/main/java/injectorpractice/service/ManufacturerServiceImpl.java:
```diff
@@ -0,0 +1,40 @@
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        manufacturerDao.update(manufacturer);
+        return manufacturer;
```
```suggestion
        return manufacturerDao.update(manufacturer);
```

AleksandrPok committed:
added empty line at the end of pom.xml;
refactored streams;
added access modifier for field manufacturerDao in ManufacturerServiceImpl class;
refactored create() and update() methods in ManufacturerServiceImpl class;
removed id from constructor for Car, Driver and Manufacturer classes.
